tests/CarbonInterval/ToDateIntervalTest.php

```diff
@@ -15,7 +15,7 @@ public function testConvertToDateInterval()
     {
         $interval = CarbonInterval::days(5)->hours(3)->minutes(50)->microseconds(123456)->invert()->toDateInterval();
 
-        $this->assertSame('DateInterval', get_class($interval));
+        $this->assertInstanceOf(DateInterval::class, $interval);
```

Comment:
Please revert this and the similar, it's not heuristic.

```php
'DateInterval' === get_class(new CarbonInterval()) // false
new CarbonInterval() instanceof DateInterval // true
```

Reply:
Thanks for the look  @kylekatarnls 
I feel the previous assertion was actually a bit misleading and would love to enforce a bit more meaning
What do you think of `$this->assertTrue(is_a($interval, DateInterval::class));` ?

Reply:
`is_a` has the same problem, it also makes the check pointless, where checking that we got a raw `DateInterval` and we don't want any kind of sub-class of `DateInterval`.

It's not a matter of code style, this test must succeed with a `DateInterval` instance and fail with a `CarbonInterval` or other classes that may extends `DateInterval`.

Reply:
> `is_a` has the same problem

I don't think so
```php
$this->assertInstanceOf(DateInterval::class, $interval);
$this->assertTrue(is_a($interval, DateInterval::class));
$this->assertFalse(is_a($interval, CarbonInterval::class));
```

Reply:
Nevertheless, I do not really care

Reply:
See c99a2c1bd8355c41f875abaeac0b5b32925241df

Reply:
Again, `is_a(new CarbonInterval(), DateInterval::class)` returns true, which is wrong regarding the meaning of the test.